packages/shared/src/contexts/UITaskManager.tsx

```diff
@@ -46,6 +62,27 @@ export const createUITaskManager = <TaskOptions extends BaseDialogProps<Result>,
                         resolve(null)
                         signal.removeEventListener('abort', abortHandler)
                     })
+
+                    // #region Mui Popper
+                    if (options?.anchorEl) {
+                        let element: HTMLElement
+                        if (options?.anchorEl instanceof HTMLElement) {
+                            element = options?.anchorEl
+                        } else {
+                            element = options?.anchorEl.currentTarget
+                        }
+
+                        // when the essential content of currentTarget would be closed over,
+                        //  we can set the anchorEl with currentTarget's bottom sibling to avoid it.
+                        const finalAnchor = options?.anchorSibling
+                            ? (element.nextElementSibling as HTMLElement) ?? undefined
+                            : element
+                        setAnchorEl(finalAnchor)
+                        // HACK: it seems like anchor doesn't work correctly
+                        // but a force repaint can solve the problem.
+                        window.requestAnimationFrame(update)
+                    }
```

Comment:
Let's keep `TaskManagerProvider` generic, move these into SelectGasSettingsMenu.

Reply:
I think `TaskManagerProvider` is not generic, it only serves `<InjectedDialog />` before this PR.